x-pack/plugins/index_management/common/constants/plugin.js

```diff
@@ -5,5 +5,12 @@
  */
 
 export const PLUGIN = {
-  ID: 'index_management'
+  ID: 'index_management',
+  VALID_LICENSE_MODES: [
+    'trial',
```

Comment:
This PR can be tested by:
1. Running with a trial license and editing this file to remove `trial` license
2. View index management page, a toast error message about license should appear.
3. View license management page, there should be no errors